src/components/implementation/no_interface/vkernel/vkernel.c

```diff
@@ -0,0 +1,195 @@
+#include <cos_component.h>
+#include <cobj_format.h>
+#include <cos_kernel_api.h>
+
+#undef assert
+#define assert(node) do { if (unlikely(!(node))) { debug_print("assert error in @ "); *((int *)0) = 0; } } while (0)
+#define PRINT_FN prints
+#define debug_print(str) (PRINT_FN(str __FILE__ ":" STR(__LINE__) ".\n"))
+#define BUG() do { debug_print("BUG @ "); *((int *)0) = 0; } while (0);
+
+struct vms_info {
+	struct cos_compinfo cinfo;
```

Comment:
I did go through this thought process of it being redundant but I wanted to keep the every information I've about the VM and the cinfo was initialized after I've pt, utpt, ct, comp. It was also going to avoid additional indirections that would be a lot more confusing especially ones like `vms_info[x].cinfo.mi.pgtbl_cap`. 
And removing these 4 in vms_info struct would break a bit of consistency. 
So I included them as part of the book-keeping.